internal/cel/library/sigstore.go

```diff
@@ -0,0 +1,682 @@
+	verify "github.com/kubewarden/policy-sdk-go/pkg/capabilities/oci/verify_v2"
+)
+
+// Sigstore provides a CEL function library extension for verifying sigstore signatures of an image.
+//
+// image
+//
+// Returns a verifier builder object that can be used to build a specific verifier.
+//
+//	kw.sigstore.image(<string>) <VerifierBuilder>
+//
+// Examples:
+//
+//	kw.sigstore.image('image:latest') // returns a verifier builder for the 'image:latest' image
+//
+// annotation
+//
+// Adds an annotation to the verifier builder.
+//
+//	<VerifierBuilder>.annotation(<string>, <string>) <VerifierBuilder>
+//
+// Examples:
+//
+//	kw.sigstore.image('image:latest').annotation('foo', 'bar').annotation('baz', 'qux') // returns a verifier builder with the annotations 'foo'='bar' and 'baz'='qux'
+//
+// pubKey
+//
+// Builds a verifier that verifies the signature of an image using a set of public keys.
+//
+//	<VerifierBuilder>.pubKey(<string>) <PubKeysVerifier>
+//	<PubKeysVerifier>.pubKey(<string>) <PubKeysVerifier>
+//
+// Examples:
+//
+//	kw.sigstore.image('image:latest').pubKey('pubkey1').pubKey('pubkey2') // returns a verifier that verifies the signature of the 'image:latest' image using the public keys 'pubkey1' and 'pubkey2'
+//
+// keyless
+//
+// Builds a verifier that verifies the signature of an image using keyless signing.
+// The first argument is the issuer and the second argument is the subject.
+//
+//	<VerifierBuilder>.keyless(<string>, <string>) <KeylessVerifier>
+//	<KeylessVerifier>.keyless(<string>, <string>) <KeylessVerifier>
+//
+// Examples:
+//
+//	kw.sigstore.image('image:latest').keyless('issuer1', 'subject1').keyless('issuer2', 'subject2') // returns a verifier that verifies the signature of the 'image:latest' image using keyless signing with the keyless info 'issuer1'='subject1' and 'issuer2'='subject2'
+//
+// keylessPrefix
+//
+// Builds a verifier that verifies the signature of an image using keyless signing.
+// The first argument is the issuer and the second argument is a subject as an URL prefix.
+// The provided subject is sanitized to ensure it is a valid URL prefix and to prevent typosquatting.
+// The signature is satisfied only if the subject is a prefix of the signature subject.
+//
+//	<VerifierBuilder>.keylessPrefix(<string>, <string>) <KeylessPrefixVerifier>
+//	<KeylessPrefixVerifier>.keylessPrefix(<string>, <string>) <KeylessPrefixVerifier>
+//
+// Examples:
+//
+//	kw.sigstore.image('image:latest').keylessPrefix('issuer1', 'https://example.com/').keylessPrefix('issuer2', 'https://example.org/') // returns a verifier that verifies the signature of the 'image:latest' image using keyless signing with the keyless prefix info 'issuer1'='https://example.com/' and 'issuer2'='https://example.org/'
+//
+// github
+//
+// Builds a verifier that verifies sigstore signatures of an image using keyless signatures made via Github Actions.
+// The first argument is the owner and the second argument is the repo.
```

Comment:
The repository argument is optional, if left empty we will trust anything that has been built by the user/org